test/common/tracing/http_tracer_impl_test.cc

```diff
@@ -236,22 +236,22 @@ TEST(HttpTracerUtilityTest, IsTracing) {
 
 TEST(HttpConnManFinalizerImpl, OriginalAndLongPath) {
   const std::string path(300, 'a');
-  const std::string expected_path(128, 'a');
+  const std::string path_prefix = "http://";
+  const std::string expected_path(128 - path_prefix.length(), 'a');
   std::unique_ptr<NiceMock<MockSpan>> span(new NiceMock<MockSpan>());
 
   Http::TestHeaderMapImpl request_headers{
       {"x-request-id", "id"}, {"x-envoy-original-path", path}, {":method", "GET"}};
   NiceMock<Http::AccessLog::MockRequestInfo> request_info;
 
-  Http::Protocol protocol = Http::Protocol::Http2;
   EXPECT_CALL(request_info, bytesReceived()).WillOnce(Return(10));
   EXPECT_CALL(request_info, bytesSent()).WillOnce(Return(11));
-  EXPECT_CALL(request_info, protocol()).WillOnce(Return(protocol));
```

Comment:
this is what I was talking about. Is there some assumption somewhere that we can derive the actual scheme from other metadata?